### PR TITLE
More robusts tests for requirements_file and conda_file

### PR DESCRIPTION
**Pull request by stsewd**
Previous tests were mocking the `load_config` function, now we are testing it for real

Related to https://github.com/rtfd/readthedocs.org/pull/4379

### CODE REVIEW

File: readthedocs/rtd_tests/tests/test_config_integration.py [hunk 1]
```diff
@@ -174,31 +203,67 @@ def test_extra_requirements(self, load_config):
...
-    def test_conda(self, load_config):
-        to_find = '__init__.py'
-        load_config.side_effect = create_load({
-            'conda': {
-                'file': to_find
-            }
-        })
+    @mock.patch('readthedocs.projects.models.Project.checkout_path')
+    def test_conda_with_cofig(self, checkout_path):
+        base_path = tempfile.mkdtemp()
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_config_integration.py]: I wasn't able to reuse the `apply_fs` function (only works with pytest) or `fake_paths` as the check is very strict.

File: readthedocs/rtd_tests/tests/test_config_integration.py [hunk 2]
```diff
@@ -40,14 +44,30 @@ def inner(path=None, env_config=None):
     return inner
 
 
-@mock.patch('readthedocs.doc_builder.config.load_config')
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_config_integration.py]: I have to remove the class level mock and only used in some methods

File: readthedocs/rtd_tests/tests/test_config_integration.py [hunk 3]
```diff
@@ -191,31 +216,67 @@ def test_extra_requirements(self, load_config):
...
+        base_path = tempfile.mkdtemp()
+        checkout_path.return_value = base_path
+
+        requirements_file = 'requirements.txt'
+        self.project.requirements_file = requirements_file
+        self.project.save()
+
+        full_requirements_file = path.join(base_path, requirements_file)
+        with open(full_requirements_file, 'w') as f:
+            f.write('pip')
```
Comment (humitos) [inline readthedocs/rtd_tests/tests/test_config_integration.py]: Do we need to create this file at all for the test?

File: readthedocs/rtd_tests/tests/test_config_integration.py [hunk 4]
```diff
@@ -44,18 +48,30 @@ def inner(path=None, env_config=None):
     return inner
 
 
-@mock.patch('readthedocs.doc_builder.config.load_config')
+def create_config_file(config, file_name='readthedocs.yml', base_path=None):
+    """
+    Creates a readthedocs configuration file with name
+    ``file_name`` in ``base_path``. If ``base_path`` is not given
+    a temporal directory is created.
```
[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_config_integration.py]: haha, yeah

File: readthedocs/rtd_tests/tests/test_config_integration.py (hunk 3 above)
Comment (stsewd) [reply readthedocs/rtd_tests/tests/test_config_integration.py]: Yeah, the validation is very strict, and it checks that the file exists, I wasn't able to mock this one. But I'm using pytest for our new tests, and we don't need to use this hacks anymore

**Comment by stsewd** [comment]
Rebased and updated

**Pull request merged**